llvm/lib/Target/RISCV/MCTargetDesc/RISCVMCExpr.cpp

```diff
@@ -41,8 +41,6 @@ void RISCVMCExpr::printImpl(raw_ostream &OS, const MCAsmInfo *MAI) const {
   if (HasVariant)
     OS << '%' << getVariantKindName(getKind()) << '(';
   Expr->print(OS, MAI);
-  if (Kind == VK_RISCV_CALL_PLT)
-    OS << "@plt";
```

Comment:
We want to always emit `@plt`, not never

Reply:
As explained in the description, always emitting `@plt` would lead to `jump .L1@plt, a0`, which is invalid in assembly.

`@plt` has two purposes today: (a) redundant suffix in `call` and `jump` (b) `.word x@plt` for `R_RISCV_PLT32`.

Never emitting `@plt` ensures that compiler output always uses `@plt` for one single purpose.

Reply:
I want to circle back to this patch. `call xxx@plt` is a x86 hack (ia32 started with `R_386_PC32`, then added `@plt` for `R_386_PLT32`). Newer architectures don't use this syntax.

Reply:
After chatting out of band, I hadn't appreciated GNU as assembles `call foo` to use R_RISCV_CALL_PLT these days (https://github.com/bminor/binutils-gdb/commit/70f35d72ef04cd23771875c1661c9975044a749c), so this is no longer a concern to me. The only broken situation would be using an old (pre-2.40, or >1y) binutils, but that's probably broken for other reasons already.